anta/inventory/__init__.py

```diff
@@ -304,18 +305,19 @@ def get_inventory(
         Args:
             established_only (bool, optional): Whether or not including non-established devices in the Inventory.
                                                Default False.
-            tags (List[str], optional): List of tags to use to filter devices. Default is [default].
+            tags (List[str], optional): List of tags to use to filter devices.
 
         Returns:
             InventoryDevices: An inventory with concerned devices
         """
-        if tags is None:
-            tags = [DEFAULT_TAG]
 
         inventory_filtered_tags = InventoryDevices()
```

Comment:
```suggestion

        def _filter_devices(device):
            """
            Helper function to select the devices based on the input tags
            and the requirement for an established connection.
            """
            #  is the device filtered by tags
            if tags is not None and not(any(tag in tags for tag in device.tags)):
              return False
            # is the device filtered because session was not established
            if established_only and not device.established:
              return False
            return True
            
        return InventoryDevices(list(filter(_filter_devices, self._inventory)))
```

could replace the `any` with some set intersection even.

Reply:
Yes, but I think it is not required by the code anymore. And this code block is not here in the last commit AFAIK.

Reply:
thats a replacement for the logic of the whole function - github UI did not let me make it nice

Reply:
Ok, It is now implemented with some small arrangements